cppcache/include/geode/CacheStatistics.hpp

```diff
@@ -47,9 +47,9 @@ class LocalRegion;
  */
 class APACHE_GEODE_EXPORT CacheStatistics {
  public:
-  typedef std::chrono::system_clock::time_point time_point;
+  using time_point = std::chrono::steady_clock::time_point;
 
-  CacheStatistics() : m_lastAccessTime(0), m_lastModifiedTime(0) {}
+  CacheStatistics() = default;
```

Comment:
`default` will produce an inlined definition that will be compiled in client code for library code. This is just asking for type pruning problems, which we've seen before. This constructor should instead be moved into the source file. Default is fine for internal implementation, not for published headers.

Reply:
Quite subtle catch here. Thanks for pointing it out! :)

Reply:
> This yet another awesome move away from ACE. I can appreciate the work put into this because I have tried and failed about 3 times myself. I need a bit more time to review the remaining source but right out the gate I am worried about the `clock` usage and change from `system_clock` to `steady_clock`. While I prefer we work internally in terms of `steady_clock` if any of the time is exchanged between processes or persisted it needs to be in `system_clock`.

After looking into it, CacheStatistics expects that an system_clock::time_point. This might potentially be used by any NC user to get modification and access time. But internally this stats are used to calculate elapsed time rather than a fixed time point. So what I did is to internally store it as steady_clock::time_point and public getters converts it to system_clock::timepoint by calculating the delta between both clocks.